pkg/audit/manager.go

```diff
@@ -138,6 +154,100 @@ func (am *Manager) audit(ctx context.Context) error {
 	return am.writeAuditResults(ctx, rs, updateLists, timestamp, totalViolationsPerConstraint)
 }
 
+func (am *Manager) auditResources(ctx context.Context) (*constraintTypes.Responses, error) {
+	discoveryClient, err := discovery.NewDiscoveryClientForConfig(am.mgr.GetConfig())
+	if err != nil {
+		return nil, err
+	}
+
+	responses := constraintTypes.NewResponses()
+
+	_, serverResourceLists, err := discoveryClient.ServerGroupsAndResources()
+
+	if err != nil {
+		return responses, err
+	}
+
+	errMap := make(opa.ErrorMap)
+
+	clusterAPIResources := make(map[metav1.GroupVersion]map[string]bool)
+	for _, rl := range serverResourceLists {
+		gv := metav1.GroupVersion{
+			Group:   rl.GroupVersionKind().Group,
+			Version: rl.GroupVersionKind().Version,
+		}
+		if _, ok := clusterAPIResources[gv]; !ok {
+			clusterAPIResources[gv] = make(map[string]bool)
+		}
+		for _, resource := range rl.APIResources {
+			if !strings.Contains(resource.Name, "/") {
```

Comment:
Why are you filtering on name? The act of adding it to the map should be sufficient.

IMO it should be fine to assume that a subresource does not exist without the parent resource, especially since we only care about kinds anyway.

Reply:
Per in-person discussion - this is the only reliable way to omit subresources from the map.